src/tests/java/htsjdk/samtools/SAMRecordUnitTest.java

```diff
@@ -94,4 +95,107 @@ public void testOffsetAtRead(String cigar, int expectedReferencePos, int posInRe
             SAMRecord sam = new SAMRecordSetBuilder().addFrag("test", 0, 1, false, false, cigar, null, 2);
             Assert.assertEquals(sam.getReferencePositionAtReadPosition(posInRead), expectedReferencePos);
     }
+
+    @Test(dataProvider = "referenceAtReadData")
+    public void testDeepCopyRef(String cigar, int expectedReferencePos, int posInRead) {
+
+        SAMRecord sam = new SAMRecordSetBuilder().addFrag("test", 0, 1, false, false, cigar, null, 2);
+        testDeepCopy(sam);
+    }
+
+    @Test(dataProvider = "offsetAtReferenceData")
+    public void testDeepCopyOffset(String cigar, int posInReference, int expectedPosInRead, boolean returnLastBaseIfDeleted) {
+        SAMRecord sam = new SAMRecordSetBuilder().addFrag("test", 0, 1, false, false, cigar, null, 2);
+        testDeepCopy(sam);
```

Comment:
Rather than re-using the existing `DataProviders` and ignoring most of the parameters you get, recommend creating your own `DataProvider` that emits just what you want (ie., a `SAMRecord` or a bam file). Then re-name these test cases to better reflect what they are actually testing.